### ISSUE TITLE
AVPro will not play certain videos, seems limited by resolution

### ISSUE THREAD

**Issue by mikemeyers1** [RenderHeads/UnityPlugin-AVProVideo#373]
Hi, as discussed on Unity forum I have a Media Player playing a 1024x1024 video fine, but cannot seem to load a video 2048x3078 at all, both videos are in same format as far as I can tell.

**Your Setup (please complete the following information):**
 - Unity version: 2019.30f3
 - AVPro Video version: Most recent trial
 - Operating system version: Windows 10
 - OS: Desktop - HP Laptop 15-bs0xx (HP) - Windows 10 (10.0.0) 64bit - English
CPU: Intel(R) Core(TM) i5-7200U CPU @ 2.50GHz - 4 threads - 8107KB
GPU: Intel(R) HD Graphics 620 - Intel - Direct3D 11.0 [level 11.1] - 4053KB - 16384
 - Video specs (resolution, frame-rate, codec, file size): see bug description

**Videos**

Will email the working video and example of not working video to unitysupport@renderheads.com

### Please DO NOT LINK / ATTACH YOUR PROJECT FILES HERE
Instead email the link to us unitysupport@renderheads.com

**Comment by AndrewRH** [comment]
Thanks, appreciate your detailed report.  We will get back to you soon.

**Comment by AndrewRH** [comment]
Thanks for the video file, we're investigating.   Meanwhile - if you disable the hardware decoding option (in MediaPlayer > Platform Specific > Windows) does that help it to play?

**Comment by mikemeyers1** [comment]
Yes, the larger resolution videos now play including the 3078x2048 and 1644x1024 videos I originally was having issues with. However, playing this video files slows down runtime immensely, perhaps even more so than Unity's default video player. What is more confusing is that using AVPro I can play six 1024x1024 video files simultaneously using 6 different mediaplayer components and this runs much smoother than playing a single 2048x3072 video. For example, I used six 1024x1024 videos as each side of a skybox using AVPro mediaplayer and Unity's 6-side skybox shader. The videos played **smooth** however they **refreshed at slightly different times** after about a minute of looping, which broke the illusion of a dome and revealed the cube nature of the skybox. The solution to fix this refresh issue, in my mind, would be to use a single video in the format of a cubemap layout along with a skybox shader. But the higher resolution videos needed to make a cubemap video does not seem to play remotely smooth in with AVPro. While AVPro is playing the 1024 videos faster than Unity default player, it is not playing the larger videos any faster.

**Comment by AndrewRH** [comment]
Hi,
Thanks for the further details.  We're still looking into it on our side.
As for the videos becoming out of sync - yes this is bound to happen eventually unless code is written to try to keep them in sync.  We have an example component called PlaybackSync that is included in the AVPro Video package.  This lets you specify a master MediaPlayer and then multiple slave MediaPlayer's and tries to keep them in sync.  This is just meant as an example starting point and not necessary a completely robust solution, but perhaps it will be useful or of interest to you to test.

Also, could you check that your Intel GPU drivers are up to date?

I will let you know our findings  soon as we have a similar system to test on here.

Thanks,

**Comment by AndrewRH** [comment]
According to this
https://en.wikichip.org/wiki/intel/hd_graphics_620#Hardware_Accelerated_Video
your Intel GPU should be able to decode up to 4K...so I don't see any reason why it isn't able to decode when the Hardware Acceleration option is ticked.  Your video plays fine on our test system which is the same CPU generation, but just with a HD Graphics 630 instead of a 620.   We get playback of 30fps with your video.  So I'm not sure why it's failing for you.  I guess things to check are:
1) Check that the Video API is set to Media Foundation and not DirectShow, and make sure Low Latency option is not enabled
2) Check you have latest GPU drivers for your system
3) See if resizing the video to 3840x2160 (4K) helps
4) See if transcoding it to H.265 helps
5) Check in the console window that it's printing out that correct version number for the AVPro Video plugin (both script and plugin version)
6) Try with a blank project and a fresh import of AVPro Video to see whether that helps
7) Try with a different version of Unity - say the latest 2018 LTS version

That's all I can recommend at this point.  I would expect your video to play back fine and smooth on your system so I'm not sure why it's not.

Thanks,

**Comment by mikemeyers1** [comment]
Hi, my GPU drives are up to date. As for transcoding to H.265, does AVPro have a transcoding feature or are you suggesting I use a third party software to do this? For number 5, where specifically am I looking for the version numbers to display? In the inspector? Thanks for your help.

**Comment by mikemeyers1** [comment]
Additionally, I tried using the playback sync script and I must not be using it correctly, since it does not seem to help at all. I applied the playback script as a component to all my media players and for each script component I put the same mediaplayer object as the master player, I tested several tolerance levels from 0-30ms and I always got same result - vidoes become considerably out of sync. What is the correct way to use this script?
